tasks/scripts.mjs

```diff
@@ -30,20 +30,34 @@ export async function compile (pattern, options) {
 }
 
 /**
- * Compile JavaScript ESM to CommonJS helper
+ * Compile JavaScript helper
  *
  * @param {AssetEntry} assetEntry - Asset entry
  */
 export async function compileJavaScript ([modulePath, { srcPath, destPath, filePath }]) {
   const moduleSrcPath = join(srcPath, modulePath)
   const moduleDestPath = join(destPath, filePath ? filePath(parse(modulePath)) : modulePath)
 
-  // Create Rollup bundle
-  const bundle = await rollup({
-    input: moduleSrcPath
-  })
+  // Option 1: Rollup bundle set (multiple files)
+  // - Module imports are preserved, not concatenated
+  if (moduleDestPath.endsWith('.mjs')) {
```

Comment:
This function seems to be doing 2 different jobs in two separate code paths – would it make sense to split it into two functions (one for generating a bundle and one that preserves modules) and decide which one to call in the `compile` function instead?

Reply:
Yeah could do. Only reason I didn't is that we'll just use the new code path in `rollup@3` and delete the old

In newer versions of Rollup the `{ preserveModules }` option can also output CJS/AMD

Is it a blocker?

Reply:
Gotcha, so once we've upgraded to v3 we'll be able to simplify this?

Reply:
Exactly. In this PR we process:

1. Entry point `*.mjs` files for ESM, top level only
2. Every non-test `*.mjs` file for UMD, recursively with `**`

```js
task.name('compile:ms', () => scripts.compile('!(*.test).mjs') // Entries all.mjs, common.mjs etc
task.name('compile:js', () => scripts.compile('**/!(*.test).mjs') // Every non-test *.mjs file
```

But in future we'll use entry points for both and let Rollup discover all child modules.

Reply:
OK, that makes sense 👍🏻 Happy to leave this for now then.

Reply:
We tried this in https://github.com/alphagov/govuk-frontend/issues/3462 but we'd have to output CommonJS rather than the hybrid Universal Module Definition (UMD) bundles for browsers ([see comment](https://github.com/alphagov/govuk-frontend/issues/3462#issuecomment-1544304034))